src/test/java/uk/gov/ons/census/casesvc/service/FulfilmentRequestProcessorTest.java

```diff
@@ -0,0 +1,110 @@
+package uk.gov.ons.census.casesvc.service;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import static uk.gov.ons.census.casesvc.model.entity.EventType.FULFILMENT_REQUESTED;
+import static uk.gov.ons.census.casesvc.testutil.DataUtils.getRandomCase;
+import static uk.gov.ons.census.casesvc.testutil.DataUtils.getTestResponseManagementEvent;
+
+import java.util.Optional;
+import java.util.UUID;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+import uk.gov.ons.census.casesvc.logging.EventLogger;
+import uk.gov.ons.census.casesvc.model.dto.EventDTO;
+import uk.gov.ons.census.casesvc.model.dto.FulfilmentRequestDTO;
+import uk.gov.ons.census.casesvc.model.dto.ResponseManagementEvent;
+import uk.gov.ons.census.casesvc.model.entity.Case;
+import uk.gov.ons.census.casesvc.model.repository.CaseRepository;
+
+@RunWith(MockitoJUnitRunner.class)
+public class FulfilmentRequestProcessorTest {
+
+  @Mock private CaseProcessor caseProcessor;
+
+  @Mock private CaseRepository caseRepository;
+
+  @Mock private EventLogger eventLogger;
+
+  @InjectMocks FulfilmentRequestProcessor underTest;
+
+  @Test
+  public void testGoodFulfilmentRequest() {
+    ResponseManagementEvent managementEvent = getTestResponseManagementEvent();
```

Comment:
Still works fine, but might be worth using the tailored test object for fulfilment here?

```suggestion
    ResponseManagementEvent managementEvent = getTestResponseManagementFulfilmentRequestedEvent();
```

Reply:
Receipt and refusal tests both use the generic method call so just following the consistency!